python/pyarrow/fs.py

```diff
@@ -178,12 +178,18 @@ def _resolve_filesystem_and_path(path, filesystem=None, *, memory_map=False):
         try:
             filesystem, path = FileSystem.from_uri(path)
         except ValueError as e:
-            # neither an URI nor a locally existing path, so assume that
-            # local path was given and propagate a nicer file not found error
-            # instead of a more confusing scheme parsing error
-            if "empty scheme" not in str(e) \
-                    and "Cannot parse URI" not in str(e):
-                raise
+            try:
+                import fsspec
+            except ImportError:
+                # neither an URI nor a locally existing path, so assume that
+                # local path was given and propagate a nicer file not found error
+                # instead of a more confusing scheme parsing error
+                if "empty scheme" not in str(e) \
+                        and "Cannot parse URI" not in str(e):
+                    raise e
+            else:
+                fs, path = fsspec.url_to_fs(path)
+                filesystem = _ensure_filesystem(fs)
```

Comment:
Here we need more verbose error handling, this is just a quick draft implementation.